docs/cross-contract/callbacks.md

```diff
@@ -46,61 +46,125 @@ mod ext_calculator {
+        ext_allowlist::ext(get_allowlist_contract())
+            .with_static_gas(XCC_GAS)
+            .is_allowlisted(get_account_to_check())
+        .then(
+            Self::ext(env::current_account_id())
+            .with_static_gas(XCC_GAS)
+            .callback_arg_macro()
+        )
+    }
 ```
 
-The syntax begins with `ext_whitelist::is_whitelisted(…` showing that we're using the trait to call the method `is_whitelisted`. However, the trait shows only one argument, but we've included four. The last three arguments are used behind the scenes and represent:
+The syntax begins with `ext_allowlist::ext()` showing that we're using the trait to call the method on the account passed into `ext()`. We then use `with_static_gas()` to specify a base amount of GAS to attach to the call. We then call the method `is_allow_listed()` and pass in the parameters we'd like to attach.
 
-1. The target smart contract account. (In this case, `whitelist.demo.testnet`.)
-2. An attached deposit of Ⓝ, in yoctoⓃ. (1 Ⓝ = 1000000000000000000000000 yoctoⓃ.)
-3. The amount of gas.
+There are a couple things to note when doing these function calls:
+1. You can attach a deposit of Ⓝ, in yoctoⓃ to the call by specifying the `.with_attached_deposit()` method but it is defaulted to 0 (1 Ⓝ = 1000000000000000000000000 yoctoⓃ, or 1^24 yoctoⓃ).
+2. You can attach a static amount of GAS by specifying the `.with_static_gas()` method but it is defaulted to 0.
+3. You can attach an unused GAS weight by specifying the `.with_unused_gas_weight()` method but it is defaulted to 1. The unused GAS will be split amongst all the functions in the current execution depending on their weights. If there is only 1 function, any weight above 1 will result in all the unused GAS being attached to that function. If you specify a weight of 0, however, the unused GAS will **not** be attached to that function. If you have two functions, one with a weight of 3, and one with a weight of 1, the first function will get `3/4` of the unused GAS and the other function will get `1/4` of the unused GAS.
 
 The two methods in the snippet above are very similar, except they will call separate callbacks in the smart contract, `callback_promise_result` and `callback_arg_macro`. 
 
-```rust reference
-https://github.com/mikedotexe/cross-contract-view/blob/06bdb3222622c824b9f2fe0a53536e6914435580/src/lib.rs#L56-L75
+```rust
+#[private]
+pub fn callback_promise_result(&mut self) -> bool {
+    assert_eq!(env::promise_results_count(), 1, "ERR_TOO_MANY_RESULTS");
+    match env::promise_result(0) {
+        PromiseResult::NotReady => unreachable!(),
+        PromiseResult::Successful(val) => {
+            if let Ok(is_allowlisted) = near_sdk::serde_json::from_slice::<bool>(&val) {
+                is_allowlisted
+            } else {
+                env::panic_str("ERR_WRONG_VAL_RECEIVED")
+            }
+        },
+        PromiseResult::Failed => env::panic_str("ERR_CALL_FAILED"),
+    }
```

Comment:
I feel like this should be documented after `callback_result`. This is just the low-level and manual option of `callback_result`

Reply:
Fixed